packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -109,6 +219,210 @@ abstract contract PermissionManager is Initializable {
+        }
+
+        uint256 newFlags = permission.delegations[_delegatee] | _flags;
+        permission.delegations[_delegatee] = newFlags;
+
+        emit PermissionDelegated(_where, _permissionIdOrSelector, _delegatee, newFlags);
+    }
+
+    /// @notice Function to remove sepcific flags from the delegatee
+    /// @param _where The address of the target contract for which `_who` receives permission.
+    /// @param _permissionIdOrSelector The permission hash or function selector used for this permission.
+    /// @param _delegatee The addresses we want to undelegate specifc flags.
+    /// @param _flags The flags as uint256 the permission owner wants to remove from this specific delegatee.
+    function undelegatePermission(
+        address _where,
+        bytes32 _permissionIdOrSelector,
+        address _delegatee,
+        uint256 _flags
+    ) public {
+        if (_flags == 0) {
+            revert FlagCanNotBeZero();
+        }
+
+        bytes32 permHash = permissionHash(_where, _permissionIdOrSelector);
+        Permission storage permission = permissions[permHash];
+
+        if (!_checkFlags(permission.owners[msg.sender], _flags)) {
+            revert UnauthorizedOwner(msg.sender, permission.owners[msg.sender], _flags);
+        }
+
+        uint256 currentFlags = permission.delegations[_delegatee];
+        if (!_checkFlags(currentFlags, _flags)) {
+            revert InvalidFlagsForRemovalPassed(currentFlags, _flags);
+        }
+
+        uint256 newFlags = currentFlags ^ _flags;
+        permission.delegations[_delegatee] = newFlags;
+
+        emit PermissionUndelegated(_where, _permissionIdOrSelector, _delegatee, newFlags);
+    }
+
+    /// @notice Function to add a new owner to a permission.
+    /// @param _where The address of the target contract for which `_who` receives permission.
+    /// @param _permissionIdOrSelector The permission hash or function selector used for this permission.
+    /// @param _owner The new manager for this permission.
+    /// @param _flags The flags as uint256 to restrict what this specifc owner actually can do. (only revoke? only grant? both?)
+    function addOwner(
+        address _where,
+        bytes32 _permissionIdOrSelector,
+        address _owner,
+        uint256 _flags
+    ) external {
+        if (_flags == 0) {
+            revert FlagCanNotBeZero();
+        }
+
+        if (_owner == address(0) || _where == address(0)) {
+            revert ZeroAddress();
+        }
+
+        Permission storage permission = permissions[
+            permissionHash(_where, _permissionIdOrSelector)
+        ];
+
+        if (_isPermissionFrozen(permission)) {
+            revert PermissionFrozen(_where, _permissionIdOrSelector);
+        }
+
+        if (!_checkFlags(permission.owners[msg.sender], _flags)) {
+            revert UnauthorizedOwner(msg.sender, permission.owners[msg.sender], _flags);
+        }
```

Comment:
response: `ownerAuth` checks permission owners and root in case the permission has no owners. Only owners are allowed to add owners, root is not allowed